core/ui-session/src/session.ts

```diff
@@ -0,0 +1,386 @@
+let agentExecutor;
+// save the current account for later usage
+let lastAccount = '';
+// save the current account for later usage
+let lastIdentity = '';
+// used to check for url overwritten account / identity parameters
+const queryParams = routing.getQueryParameters();
+// used to only start runtime resolve once
+const runtimeLoadCache = { };
+
+export default class EvanSession {
+  /**
+   * Runtime, started for the current account (configured in localStorage) and it's profile
+   * identity.
+   */
+  static accountRuntime: Runtime;
+
+  /**
+   * Runtime, started for the current account (configured in localStorage) and the localStorage
+   * configured identity.
+   */
+  static identityRuntime: Runtime;
+
+  /**
+   * identity address of the activeAccount
+   */
+  static accountIdentity: string;
+
+  /**
+   * Has the account a old profile, that is bount to the account and not to the identity?
+   */
+  static isOldProfile: boolean;
+
+  static get activeAccount(): string {
+    switch (EvanSession.provider) {
+      case 'agent-executor': {
+        return agentExecutor.accountId;
+      }
+      // internal
+      default: {
+        // if the url was opened using an specific accountId, use this one!
+        if (queryParams.accountId) {
+          return queryParams.accountId;
+        }
+      }
+    }
+
+    return window.localStorage['evan-account'];
+  }
+
+  static set activeAccount(accountId: string) {
+    lastAccount = accountId;
+    window.localStorage['evan-account'] = accountId;
+  }
+
+  static get activeIdentity(): string {
+    switch (EvanSession.provider) {
+      case 'agent-executor': {
+        return agentExecutor.identity;
+      }
+      // internal
+      default: {
+        // if the url was opened using an specific identity, use this one!
+        if (queryParams.identity) {
+          return queryParams.identity;
+        }
+
+        break;
+      }
+    }
+
+    return window.localStorage['evan-identity'] || EvanSession.accountIdentity;
+  }
+
+  static set activeIdentity(identity: string) {
+    lastIdentity = identity;
+    window.localStorage['evan-identity'] = identity;
+  }
+
+  static get provider(): string {
+    let provider = 'internal';
+
+    if (agentExecutor) {
+      provider = 'agent-executor';
+    } else if (queryParams.provider) {
+      provider = queryParams.provider;
+    }
+
+    return provider;
+  }
+
+  static set provider(value: string) {
+    window.localStorage['evan-provider'] = value;
+  }
+
+  /**
+   * Checks the current url parameters if agent executor login parameters are given.
+   *
+   * @return     {any}  all agent-exeutor parameters for requesting smart-agents and decrypting the
+   *                    profile ({ accountId, agentUrl, key, token, })
+   */
+  static async getAgentExecutor(): Promise<any> {
+    // if the agentExecutor wasn't loaded before, check if the query parameter was specified
+    if (typeof agentExecutor === 'undefined') {
```

Comment:
isn't it the same as this?
```suggestion
    if (agentExecutor === undefined) {
```